packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,357 @@
+                  type: selectionVariant
+                })
+              }
+              return item
+            })
+          })
+        }
+      }
+      return acc
+    }, {})
+
+    return (
+      <div
+        className={clsx(styles.ActionMenu, disabled && styles['ActionMenu--disabled'])}
+        data-testid={testId || testIds.root}
+      >
+        {SelectButton}
+        {showMenu ? SelectOverlay : null}
+      </div>
+    )
+  }
+)
+
+type ActionMenuButtonProps = PropsWithChildren<Ref<HTMLButtonElement>> & {
+  ref?: React.RefObject<HTMLButtonElement>
+  className?: string
+  menuOpen?: boolean
+  disabled?: boolean
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
+  'data-testid'?: string
+}
+
+const ActionMenuButton = forwardRef<HTMLButtonElement, ActionMenuButtonProps>(
+  ({children, className, 'data-testid': testId, disabled, menuOpen, ...props}, ref) => {
+    return (
+      <Button
+        ref={ref}
+        className={clsx(styles.ActionMenu__button, className)}
+        hasArrow={false}
+        aria-haspopup="true"
+        aria-expanded={menuOpen ? 'true' : 'false'}
+        trailingVisual={ChevronDownIcon}
+        disabled={disabled}
+        data-testid={testId || testIds.button}
+        {...props}
+      >
+        <span className={styles['ActionMenu__button-text']}>{children}</span>
+      </Button>
+    )
+  }
+)
+
+type ActionMenuItemProps = {
+  value: string
+  handler?: (newValue: string) => void
+  selected?: boolean
+  type?: 'none' | 'single'
+  disabled?: boolean
+} & PropsWithChildren<React.HTMLProps<HTMLLIElement>>
+
+const roleTypeMap = {
+  none: 'menuitem',
+  single: 'menuitemradio'
+  // multiple: 'menuitemcheckbox'
+}
+
+const ActionMenuItem = ({
+  children,
+  className,
+  disabled,
+  handler,
+  selected,
+  type,
+  value,
+  ...props
+}: ActionMenuItemProps) => {
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
+    <li
+      className={clsx(styles.ActionMenu__item, className)}
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
+      role={roleTypeMap[type || 'single']}
+      aria-checked={type === 'none' ? undefined : selected ? 'true' : 'false'}
+      aria-disabled={disabled ? 'true' : 'false'}
+      onClick={handler && !disabled ? () => handler(value) : undefined}
+      tabIndex={disabled ? undefined : 0}
+      {...props}
+    >
+      {type !== 'none' && (
+        <span className={styles['ActionMenu__item-leadingVisual']}>
+          {selected && (
+            <CheckIcon
+              className={clsx(styles['ActionMenu__item-icon'], disabled && styles['ActionMenu__item-icon--disabled'])}
+              size={16}
+            />
+          )}
+        </span>
+      )}
+
+      <span className={styles['ActionMenu__item-text']}>
+        <Text
+          variant={disabled ? 'muted' : 'default'}
```

Comment:
are `disabled` items necessary for Brand purposes?

Reply:
I can see this being useful in future. Particularly with dynamic data, where we would otherwise need to re-render a filtered set of data which is worse for accessibility and ux. Can we support this in Figma too?